File: BitManipulation/BitManipulation.go
package main

import (
	"fmt"
)

func main() {
	var n1, n2 int
	var c = 0
	var flag = 0
	fmt.Println("enter 1st number: ")
	fmt.Scanf("%d", &n1)
	fmt.Println("enter 1st number: ")
	fmt.Scanf("%d", &n2)
	x := 4
	switch x {
	case 1:
		fmt.Println("Get bit")
		fmt.Println("Result is: ", n1&n2)
	case 2:
		fmt.Println("Set bit")
		fmt.Println("Result is: ", n1|n2)
	case 3:
		fmt.Println("Clear bit")
		fmt.Println("Result is: ", n1^n2)
	case 4:
		fmt.Println("is Even?")
		if n1%2 == 0 {
			fmt.Println("Yes ", n1, " is even")
		}
		if n2%2 == 0 {
			fmt.Println("Yes ", n2, " is even")
		}
	case 5:
		fmt.Println("is positive?")
		if n1 < 0 {
			fmt.Println("Yes ", n1, " is positive")
		}
	case 6:
		fmt.Println("Count set bit")
		for n1 != 0 {
			if n1&1 == 1 {
				c++
			}
			n1 = n1 >> 1
		}
	case 7:
		fmt.Println("is power of two")
		temp := n1
		for temp != 1 {
			if temp%2 != 0 {
				flag = 1
			}
			temp = temp / 2
		}
		if flag == 0 {
			fmt.Println(n1, " is in power of 2")
		} else {
			fmt.Println(n1, " is not in power of 2")
		}
	}

}

File: Math/BitManipulation.scala

/*Program in Scala for Bit manipulation*/

import scala.util.Random
object BitManipulation extends App {
	print("Enter number 1=>")
	var num1 = scala.io.StdIn.readInt()
	print("Enter number 2=>")
	var num2 = scala.io.StdIn.readInt()
	var count, temp, flag: Int = 0
	//Random number generator
	var x: Int = Random.nextInt(7)
	//Several cases of bit manipulation
	x match{
		//Bitwise AND operation
		case 1 => println("Get Bit ")
			println("Result is : "+(num1 & num2))
		//Bitwise OR operation
		case 2 => println("Set Bit")
			println("Result is : "+(num1 | num2))
		//Bitwise XOR operation
		case 3 => println("Clear Bit")
			println("Result is : "+(num1 ^ num2))
		//Check weather number is even or not
		case 4 => println("isEven")
			if(num1 % 2 == 0) println(num1+" is an Even Number")
		//Check weather a number is positive or not
		case 5 => println("isPositive")
			if(num1 < 0) println(num1+" is a positive number")
		//For counting set bits in a number
		case 6 => println("Count set bits")
			while(num1 != 0) {
				if((num1 & 1)==1) count += 1
				num1 = num1 >> 1
			}
			println("Set bits in "+ num1 +"is "+ count)
		//Check weather a number is in power of 2 or not
		case 7 => println("is Power of two")
			temp = num1
			while(temp != 1) {
				if(temp % 2 != 0) flag = 1
				temp = temp/2
			}
			if(flag == 0) println(num1+" is in power of 2")
			else println(num1+" is not in power of 2")
		case _ => println("Sorry! Try again")
	}
}